app/lib/account_statuses_filter.rb

```diff
@@ -60,8 +60,8 @@ def filtered_reblogs_scope
       .where(reblog_of_id: nil)
       .or(
         scope
+          .where(reblog: { accounts: { domain: nil } }).or(scope.where.not(reblog: { accounts: { domain: current_account.excluded_from_timeline_domains } }))
           .where.not(reblog: { account_id: current_account.excluded_from_timeline_account_ids })
-          .where.not(reblog: { accounts: { domain: current_account.excluded_from_timeline_domains } })
```

Comment:
To reduce code duplication and improve legibility, I'd like to use the `Status.not_excluded_by_account` and `Status.not_domain_blocked_by_account` scopes… however, I don't know if this is possible to do specifically on `reblog`.